pallets/runtime/common/src/runtime.rs

```diff
@@ -725,34 +725,41 @@ macro_rules! runtime_apis {
                 }
             }
 
-            /*
-            impl pallet_contracts_rpc_runtime_api::ContractsApi<Block, polymesh_primitives::AccountId, Balance, BlockNumber>
-                for Runtime
-            {
+            impl pallet_contracts_rpc_runtime_api::ContractsApi<
+                Block,
+                polymesh_primitives::AccountId,
+                Balance,
+                BlockNumber,
+                polymesh_primitives::Hash,
+            > for Runtime {
                 fn call(
                     origin: polymesh_primitives::AccountId,
                     dest: polymesh_primitives::AccountId,
                     value: Balance,
                     gas_limit: u64,
                     input_data: Vec<u8>,
-                ) -> ContractExecResult {
-                    BaseContracts::bare_call(origin, dest.into(), value, gas_limit, input_data)
+                ) -> pallet_contracts_primitives::ContractExecResult {
+                    Contracts::bare_call(origin, dest, value, gas_limit, input_data, false)
+                }
+
+                fn instantiate(
+                    origin: polymesh_primitives::AccountId,
+                    endowment: Balance,
+                    gas_limit: u64,
+                    code: pallet_contracts_primitives::Code<polymesh_primitives::Hash>,
+                    data: Vec<u8>,
+                    salt: Vec<u8>,
+                ) -> pallet_contracts_primitives::ContractInstantiateResult<polymesh_primitives::AccountId> {
+                    Contracts::bare_instantiate(origin, endowment, gas_limit, code, data, salt, false)
```

Comment:
I think this RPC endpoint needs the logic from our contract pallet's `instantiate_with_code`, (i.e. attach the contract address to the caller's identity).

Reply:
I think these RPC endpoints are only for testing if an instantiate/call will work (i.e. dry-run).  It should be fine to use the `instantiate` call without permissions.  We just want this interface to be as compatible with the Contracts UI as possible.